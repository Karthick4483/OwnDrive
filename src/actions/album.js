// @flow
/**
 * @module Actions/Album
 * @desc Album Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const {
  getAlbums,
  deleteAlbum,
  trashAlbum,
  createAlbum,
  addFilesToAlbum,
  removeFilesFromAlbum,
  getAlbumFiles,
} = createActions({
  [ActionTypes.GET_ALBUMS]: params => params,
  [ActionTypes.DELETE_ALBUM]: params => params,
  [ActionTypes.TRASH_ALBUM]: params => params,
  [ActionTypes.CREATE_ALBUM]: params => params,
  [ActionTypes.GET_ALBUM_FILES]: params => params,
  [ActionTypes.REMOVE_ALBUM_FILES]: params => params,
  [ActionTypes.ADD_FILES_TO_ALBUM]: params => params,
  [ActionTypes.REMOVE_FILES_FROM_ALBUM]: params => params,
});
