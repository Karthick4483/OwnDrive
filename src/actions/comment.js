// @flow
/**
 * @module Actions/Comment
 * @desc Comment Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { getFileComments, deleteFileComment, addFileComment } = createActions({
  [ActionTypes.GET_FILE_COMMENTS]: params => params,
  [ActionTypes.DELETE_FILE_COMMENT]: params => params,
  [ActionTypes.ADD_FILE_COMMENT]: params => params,
});
