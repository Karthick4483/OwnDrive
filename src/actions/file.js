// @flow
/**
 * @module Actions/File
 * @desc File Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const {
  getFiles,
  getPhotos,
  getTrashFiles,
  deleteFiles,
  moveFiles,
  uploadFiles,
  trashFiles,
  createFolder,
  restoreFiles,
} = createActions({
  [ActionTypes.GET_FILES]: params => params,
  [ActionTypes.GET_PHOTOS]: params => params,
  [ActionTypes.GET_TRASH_FILES]: () => ({}),
  [ActionTypes.DELETE_FILES]: (id, path) => ({ id, path }),
  [ActionTypes.MOVE_FILES]: data => data,
  [ActionTypes.TRASH_FILES]: (id, path) => ({ id, path }),
  [ActionTypes.RESTORE_FILES]: (id, path) => ({ id, path }),
  [ActionTypes.UPLOAD_FILES]: formData => ({ formData }),
  [ActionTypes.CREATE_FOLDER]: data => data,
});
