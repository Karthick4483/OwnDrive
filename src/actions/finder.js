// @flow
/**
 * @module Actions/Finder
 * @desc Finder Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { getFinderFiles } = createActions({
  [ActionTypes.GET_FINDER_FILES]: params => params,
});
