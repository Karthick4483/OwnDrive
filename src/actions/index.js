export * from './app';
export * from './user';
export * from './file';
export * from './finder';
export * from './comment';
export * from './album';
