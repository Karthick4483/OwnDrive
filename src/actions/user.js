// @flow
/**
 * @module Actions/User
 * @desc User Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { userLogin: login } = createActions({
  [ActionTypes.USER_LOGIN]: () => ({}),
});

export function logOut() {
  window.location.href = '/login';
}
