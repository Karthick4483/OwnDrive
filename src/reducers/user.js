import { handleActions } from 'redux-actions';
import immutable from 'immutability-helper';

import { STATUS, ActionTypes } from 'constants/index';

export const userState = {
  isAuthenticated: true,
  status: STATUS.IDLE,
  data: {},
};

export default {
  user: handleActions(
    {
      [ActionTypes.USER_LOGIN]: state =>
        immutable(state, {
          status: { $set: STATUS.RUNNING },
        }),
      [ActionTypes.USER_LOGIN_SUCCESS]: state =>
        immutable(state, {
          isAuthenticated: { $set: true },
          status: { $set: STATUS.READY },
        }),
      [ActionTypes.USER_LOGOUT]: state =>
        immutable(state, {
          status: { $set: STATUS.RUNNING },
        }),
      [ActionTypes.USER_LOGOUT_SUCCESS]: state =>
        immutable(state, {
          isAuthenticated: { $set: false },
          status: { $set: STATUS.IDLE },
        }),
      [ActionTypes.INIT_USER]: state =>
        immutable(state, {
          isAuthenticated: { $set: false },
          status: { $set: STATUS.RUNNING },
        }),
      [ActionTypes.INIT_USER_SUCCESS]: (state, response) =>
        immutable(state, {
          isAuthenticated: { $set: true },
          status: { $set: STATUS.READY },
          data: { $set: response.payload.data },
        }),
      [ActionTypes.INIT_USER_FAILURE]: state =>
        immutable(state, {
          isAuthenticated: { $set: false },
          status: { $set: STATUS.IDLE },
        }),
    },
    userState,
  ),
};
