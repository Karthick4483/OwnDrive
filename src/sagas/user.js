/**
 * @module Sagas/User
 * @desc User
 */

import { all, delay, call, put, takeLatest } from 'redux-saga/effects';
import { request } from 'modules/client';

import { ActionTypes } from 'constants/index';
import { getFiles, getTrashFiles } from './file';

/**
 * Login
 */
export function* login() {
  try {
    yield delay(400);

    yield put({
      type: ActionTypes.USER_LOGIN_SUCCESS,
    });
  } catch (err) {
    /* istanbul ignore next */
    yield put({
      type: ActionTypes.USER_LOGIN_FAILURE,
      payload: err,
    });
  }
}

/**
 * Logout
 */
export function* logout() {
  try {
    yield call(request, '/api/auth/logout');
    window.location.reload();
    yield put({
      type: ActionTypes.USER_LOGOUT_SUCCESS,
    });
  } catch (err) {
    /* istanbul ignore next */
    yield put({
      type: ActionTypes.USER_LOGOUT_FAILURE,
      payload: err,
    });
  }
}

export function* initUser() {
  try {
    const response = yield call(request, '/api/user/me');
    yield put({
      type: ActionTypes.INIT_USER_SUCCESS,
      payload: { data: response },
    });
    yield call(getFiles);
    yield call(getTrashFiles);
  } catch (err) {
    /* istanbul ignore next */
    yield put({
      type: ActionTypes.INIT_USER_FAILURE,
      payload: err,
    });
  }
}

/**
 * User Sagas
 */
export default function* root() {
  yield all([
    takeLatest(ActionTypes.INIT_USER, initUser),
    takeLatest(ActionTypes.USER_LOGOUT, logout),
  ]);
}
