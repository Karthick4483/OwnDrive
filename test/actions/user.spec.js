import { userLogin, userLogout } from 'actions/user';

describe('App', () => {
  it('login should return an action', () => {
    expect(userLogin()).toMatchSnapshot();
  });

  it('logOut should return an action', () => {
    expect(userLogout()).toMatchSnapshot();
  });
});
