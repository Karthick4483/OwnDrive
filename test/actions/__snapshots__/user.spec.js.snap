// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App logOut should return an action 1`] = `undefined`;

exports[`App login should return an action 1`] = `
Object {
  "payload": Object {},
  "type": "USER_LOGIN",
}
`;
